### PR TITLE
adds trigger name to tigger_fire event

### PR DESCRIPTION
**Pull request by jahilliard**
## Changes in this pull request

- Adds trigger name to tigger_fire event 
Screen Shot 2022-04-21 at 3 44 20 PM

### Checklist

- [x] All tests pass. Demo project builds and runs.
- [ ] I added tests, an experiment, or detailed why my change isn't tested.
- [ ] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/superwall-me/paywall-ios/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by jakemor** [comment]
Looks good! Great move passing in EventData as opposed to just the name 

- need tests for when eventData is provided
- have you though about passing through all the event's params, not just the name?

File: Sources/Paywall/Paywall/Events/InternalEvents.swift [hunk 1]
```diff
@@ -274,23 +274,28 @@ extension Paywall {
...
       )
-    case .triggerFire(let triggerInfo):
+    case let .triggerFire(triggerInfo, eventData):
       var params: [String: Any] = [:]
       switch triggerInfo {
       case .noRuleMatch:
-        params = ["result": "no_rule_match"]
+        params = [
+          "result": "no_rule_match",
+          "triggerName": eventData.name
```
Comment (anglinb) [inline Sources/Paywall/Paywall/Events/InternalEvents.swift:283]: Everything else is snake case, probably worth following the convention

Comment (jahilliard) [reply Sources/Paywall/Paywall/Events/InternalEvents.swift:283]: Friendly heads up not everything is snakecase, but we definitely should follow a single convention.  

Screen Shot 2022-04-21 at 7 10 19 PM

Comment (jahilliard) [reply Sources/Paywall/Paywall/Events/InternalEvents.swift:283]: I'll create a ticket for @yusuftor .

**Comment by jahilliard** [comment]
@jakemor 
> have you though about passing through all the event's params, not just the name?

I thought about this.  I don't think it makes sense, because we already get all the rest of the event's metadata in the original `Paywall.track` or `Paywall.trigger` call.  I don't think there is a need to duplicate data. :)

File: Sources/Paywall/Paywall/Events/InternalEvents.swift (hunk 1 above)
Comment (jahilliard) [reply Sources/Paywall/Paywall/Events/InternalEvents.swift:283]: https://linear.app/superwall/issue/SW-863/standardize-all-internalevent-properties-to-snake-case

**Comment by yusuftor** [comment]
Thanks for doing this Justin. Unfortunately I'm going to have to close this PR rather than merge because the way we do internal tracking will change. Sorry!! But I have just manually transferred your edits in to the relevant parts in my branch to make sure we include `trigger_name` in the params, so it has been useful!

**Pull request closed without merging**